### PR TITLE
change default JVM min max for ES

### PR DESCRIPTION
**Pull request by pfmackin**
The JVM args for ES nodes (master, ingest, data) were hardcoded and there was no way to change them.  I modified the code to derive them from the pod.request for each of the 3 ES node types.  The JVM value is calculated to be 75% of the pod request size to account for overhead.  The JVM min/max must be identical or else ES pods won't start.  

Also, I adjusted the default master from 512m to 700m to be conservative.

### CODE REVIEW

File: pkg/util/memory/jvmheap.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	}
+	if sizeB%UnitK == 0 {
+		// e.g. 1500k - value in exact kb
+		return fmt.Sprintf("%.0fk", float64(sizeB)/UnitK)
+	}
+	// round up to next kb
+	return fmt.Sprintf("%vk", sizeB/UnitK+1)
+}
+
+//func FormatJvmHeapSize(sizeB int64) string {
```
Comment (michael-cico) [inline pkg/util/memory/jvmheap.go:53]: Cruft?

**Review by michael-cico — DISMISSED** [review]

File: pkg/util/memory/jvmheap.go (hunk 1 above)
Comment (pfmackin) [reply pkg/util/memory/jvmheap.go]: fixed

**Review by michael-cico — APPROVED** [review]

File: pkg/util/memory/jvmheap.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+const UnitK = 1024
+
+// UnitM is 1 Megabyte
+const UnitM = 1024 * UnitK
+
+// UnitG is 1 Gigabyte
+const UnitG = 1024 * UnitM
+
+// FormatJvmHeapMinMax returns the identical min and max JVM heap setting in the format
+// java expects like "-Xms2g -Xmx2g"
```
Comment (markxnelson) [inline pkg/util/memory/jvmheap.go:20]: Java has a capital J

File: pkg/util/memory/jvmheap.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+func FormatJvmHeapMinMax(heap string) string {
+	return fmt.Sprintf("-Xms%s -Xmx%s", heap, heap)
+}
+
+// FormatJvmHeapSize formats the string based on the size of the input value
+// Return whole number (1200M not 1.2G)
+// Return min magnatude of K
```
Comment (markxnelson) [inline pkg/util/memory/jvmheap.go:27]: spelling

File: pkg/util/memory/jvmheap.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+func FormatJvmHeapSize(sizeB int64) string {
+	if sizeB >= UnitG {
+		if sizeB%UnitG == 0 {
+			// e.g 50g
```
Comment (markxnelson) [inline pkg/util/memory/jvmheap.go:31]: e.g.

File: pkg/util/memory/jvmheap.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+			return fmt.Sprintf("%.0fg", float64(sizeB)/UnitG)
+		}
+		if sizeB%UnitM == 0 {
+			// e.g. 1500m - value in exact mb
```
Comment (markxnelson) [inline pkg/util/memory/jvmheap.go:35]: mb, kb, etc., are captialized

**Review by markxnelson — COMMENTED** [review]
couple of small issues with spelling/caps/punctuation - code is fine

**Review by markxnelson — APPROVED** [review]

File: pkg/util/memory/jvmheap.go (hunk 5 above)
Comment (pfmackin) [reply pkg/util/memory/jvmheap.go:35]: either one works for JVM heap args, VMO was using lower case so i used that.

File: pkg/util/memory/jvmheap.go (hunk 2 above)
Comment (pfmackin) [reply pkg/util/memory/jvmheap.go:20]: fixed

File: pkg/util/memory/jvmheap.go (hunk 4 above)
Comment (pfmackin) [reply pkg/util/memory/jvmheap.go:31]: fixed

File: pkg/util/memory/jvmheap.go (hunk 3 above)
Comment (pfmackin) [reply pkg/util/memory/jvmheap.go:27]: fixed

**Review by michael-cico — APPROVED** [review]

**Pull request merged**